File: src/main/python/ArchiveRecord.py
# Archives Unleashed Toolkit (AUT):
# An open-source platform for analyzing web archives.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

from ExtractDate import ExtractDate, DateComponent

class ArchiveRecord(object):
  def __init__(self, contentBytes, contentString, crawlDate, crawlMonth, 
    domain, imageBytes, mimeType, url):
    self.contentBytes = contentBytes
    self.contentString = contentString
    self.crawlDate = contentString
    self.crawlMonth = crawlMonth
    self.domain = domain
    self.imageBytes = imageBytes
    self.mimeType = mimeType
    self.url = url

  def getCrawlDate(self): 
    return self.crawlDate

  def getMimeType(self):
    return self.mimeType

  def getUrl(self):
    return self.url 

  def getContentBytes(self):
    return self.contentBytes

  def getCrawlMonth(self):
    return ExtractDate(self.crawlDate, DateComponent.YYYYMM)

  def getDomain(self):
    return self.domain

  def getContentBytes(self):
    return self.contentBytes

  def getContentString(self):
    return self.contentString

  def getImageBytes(self):
    return self.imageBytes

File: src/main/scala/io/archivesunleashed/ArchiveRecord.scala
/*
 * Copyright © 2017 The Archives Unleashed Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package io.archivesunleashed

/** Trait for a record in a web archive. */
trait ArchiveRecord extends Serializable {

  /** Returns the full path or url containing the Archive Records. */
  def getArchiveFilename: String

  /** Returns the crawl date. */
  def getCrawlDate: String

  /** Returns the crawl month. */
  def getCrawlMonth: String

  /** Returns the content of the record as a String. */
  def getContentString: String

  /** Returns the MIME type. */
  def getMimeType: String

  /** Returns the URL. */
  def getUrl: String

  /** Returns the domain. */
  def getDomain: String

  /** Returns a raw array of bytes for an image. */
  def getBinaryBytes: Array[Byte]

  /** Returns the http status of the crawl. */
  def getHttpStatus: String

  /** Returns payload digest (SHA1). */
  def getPayloadDigest: String
}
